app.js

```diff
@@ -2,22 +2,120 @@
   if (!window.addEventListener) return // Check for IE9+
 
   let options = INSTALL_OPTIONS
+  let updateTimeout
+  const delay = 1500
+  const elements = []
+  const PAYPAL_SCRIPT_URL = "https://cdn.rawgit.com/paypal/JavaScriptButtons/master/dist/button.js"
+  const TIME_PERIOD_SYMBOLS = {
+    D: "Daily",
+    W: "Weekly",
+    M: "Monthly",
+    Y: "Yearly"
+  }
+
+  function updateElements() {
+    if (options.merchant) {
+      const {buttons} = options
+
+      buttons
+        .filter($ => $.name && $.amount)
```

Comment:
I think you were right on our first approach. The filter is a bit weird since the user won't get any feedback until they fill all the fields.